tests/subsys/bluetooth/gatt_dm/src/main.c

```diff
@@ -467,29 +469,10 @@ void test_gatt_many_serv_by_uuid(void)
 		      bt_gatt_dm_attr_cnt(dm),
 		      "Unexpected number of attributes detected: %d",
 		      bt_gatt_dm_attr_cnt(dm));
-}
 
-void test_main(void)
-{
-	ztest_test_suite(
-		test_gatt,
-		ztest_unit_test_setup_teardown(test_gatt_none_serv, test_setup, unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_DIS_simple_next_attr, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_DIS_attr_by_handle, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_HIDS_simple_next_attr, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_HIDS_attr_by_handle, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_HIDS_next_chrc_access, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_HIDS_chrc_by_uuid, test_setup,
-					       unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_generic_serv, test_setup, unit_test_noop),
-		ztest_unit_test_setup_teardown(test_gatt_many_serv_by_uuid, test_setup,
-					       unit_test_noop)
-	);
-
-	ztest_run_test_suite(test_gatt);
+	/* ------------------------------------------------------ */
+	/* Clean up */
+	bt_gatt_dm_data_release(dm);
+	zassert_equal(0, bt_gatt_dm_attr_cnt(dm), "Parameter count after clearing: %d",
+		      bt_gatt_dm_attr_cnt(dm));
```

Comment:
Is it possible to move the clean-up to the `test_after` function?

Reply:
No really. We have at least one test case that no required clearing and return NULL pointer to the dm